app/src/main/java/de/danoeh/antennapod/fragment/CoverFragment.java

```diff
@@ -159,6 +170,35 @@ private void displayCoverImage(int position) {
         }
     }
 
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+
+        configureForOrientation(newConfig);
+    }
+
+    public float convertDpToPixel(float dp) {
+        Context context = this.getActivity().getApplicationContext();
+        return dp * ((float) context.getResources().getDisplayMetrics().densityDpi / DisplayMetrics.DENSITY_DEFAULT);
+    }
+
+    private void configureForOrientation(Configuration newConfig) {
+        LinearLayout mainContainer = getView().findViewById(R.id.cover_fragment);
+        if (newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
+            mainContainer.setOrientation(LinearLayout.VERTICAL);
+            if (newConfig.screenWidthDp > 0) {
+                imgvCover.getLayoutParams().width = (int) (convertDpToPixel(newConfig.screenWidthDp) * .80);
+                imgvCover.getLayoutParams().height = imgvCover.getLayoutParams().width;
```

Comment:
Setting this directly just sets a variables and does not re-layout. I think something like this would be more safe:

```
ViewGroup.LayoutParams params = imgvCover.getLayoutParams();
params.width = (int) (convertDpToPixel(newConfig.screenWidthDp) * .80);
params.height = imgvCover.getLayoutParams().width;
imgvCover.setLayoutParams(params);
```

Reply:
Done